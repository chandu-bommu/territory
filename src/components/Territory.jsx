import React, { Component } from 'react';
import '../style/App.css';
import { PrimaryButton } from 'office-ui-fabric-react/lib/Button';
import { Label } from 'office-ui-fabric-react/lib/Label';
import { Dropdown, DropdownMenuItemType } from 'office-ui-fabric-react/lib/Dropdown';


class Territory extends React.Component{

  constructor(props) {
    super(props);
    this.state = {
      selectedItem: undefined,
      selectedItems: []
    };
  }

  componentWillMount = () => {}

  render = () => {
    return (
      <div className="App-intro">
        <table>
            <tr>
              <td>
                <Dropdown
                  placeHolder='Select a Country'
                  label='COUNTRY'
                  id='Basicdrop1'
                  ariaLabel='Basic dropdown example'
                  options={
                    [
                      { key: 'A', text: 'Option a' }
                    ]
                  }
                  componentRef={ this._basicDropdown }
                />
              </td>
              <td>
                <Dropdown
                  placeHolder='Select a State'
                  label='STATE'
                  id='Basicdrop1'
                  ariaLabel='Basic dropdown example'
                  options={
                    [
                      { key: 'A', text: 'Option a' }
                    ]
                  }
                  componentRef={ this._basicDropdown }
                />
              </td>
              <td>
                <Dropdown
                  placeHolder='Select a City'
                  label='CITY'
                  id='Basicdrop1'
                  ariaLabel='Basic dropdown example'
                  options={
                    [
                      { key: 'A', text: 'Option a' }
                    ]
                  }
                  componentRef={ this._basicDropdown }
                />
              </td>
              <td>
              <Dropdown
                placeHolder='Select Areas'
                selectedKeys={ this.state.selectedItems }
                onChanged={ this.onChangeMultiSelect }
                onFocus={ this._log('onFocus called') }
                onBlur={ this._log('onBlur called') }
                multiSelect
                options={
                  [
                    { key: 'Header4', text: 'Colors', itemType: DropdownMenuItemType.Header },
                    { key: 'red', text: 'Red' },
                    { key: 'green', text: 'Green' },
                    { key: 'blue', text: 'Blue' },
                    { key: 'yellow', text: 'Yellow' },
                    { key: 'divider_2', text: '-', itemType: DropdownMenuItemType.Divider },
                    { key: 'Header5', text: 'Flower', itemType: DropdownMenuItemType.Header },
                    { key: 'rose', text: 'Rose' },
                    { key: 'lily', text: 'Lily' },
                    { key: 'sunflower', text: 'Sunflower' },
                  ]
                }
              />
              </td>
              <td>
                <div>
                  <Label>Click the button to insert text.</Label>
                  <PrimaryButton
                    data-automation-id='test'
                    text='Add to Table'
                  />
                </div>
              </td>
            </tr>
        </table>
      </div>
    );
  }

  componentDidMount = () => {}

}

export default Territory;
