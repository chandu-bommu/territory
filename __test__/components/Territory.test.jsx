import React from 'react';
import Territory from '../../src/components/Territory.jsx';
import { shallow, mount } from 'enzyme';

describe('Territory', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Territory />);
    expect(wrapper).toMatchSnapshot();
  });
});
