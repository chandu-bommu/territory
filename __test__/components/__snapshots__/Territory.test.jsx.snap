// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Territory renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="appComponent"
  >
    <div
      className="countryForm"
    >
      <table>
        <thead>
          <tr>
            <th>
              COUNTRY
            </th>
            <th>
              STATE
            </th>
            <th>
              CITY
            </th>
            <th>
              AREAS
            </th>
            <th />
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              <WithResponsiveMode
                id="countryDrop"
                onChanged={[Function]}
                options={Array []}
                placeHolder="Select a Country"
              />
            </td>
            <td>
              <WithResponsiveMode
                id="stateDrop"
                onChanged={[Function]}
                options={Array []}
                placeHolder="Select a State"
              />
            </td>
            <td>
              <WithResponsiveMode
                id="cityDrop"
                onChanged={[Function]}
                options={Array []}
                placeHolder="Select a City"
              />
            </td>
            <td>
              <WithResponsiveMode
                multiSelect={true}
                onChanged={[Function]}
                options={Array []}
                placeHolder="Select Areas"
              />
            </td>
            <td>
              <CustomizedDefaultButton
                data-automation-id="resetButton"
                onClick={[Function]}
                text="Reset"
              />
            </td>
          </tr>
          <tr>
            <td
              colSpan="5"
              style={
                Object {
                  "textAlign": "right",
                }
              }
            >
              <CustomizedPrimaryButton
                data-automation-id="addToTableButton"
                id="addButton"
                onClick={[Function]}
                text="Add to Table"
              />
              <CustomizedPrimaryButton
                data-automation-id="clearTableButton"
                id="clearButton"
                onClick={[Function]}
                text="Clear Table"
              />
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      className="countryTable"
    >
      <table>
        <thead>
          <tr>
            <th>
              Country
            </th>
            <th>
              State
            </th>
            <th>
              City
            </th>
            <th>
              Areas
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>,
  "nodes": Array [
    <div
      className="appComponent"
    >
      <div
        className="countryForm"
      >
        <table>
          <thead>
            <tr>
              <th>
                COUNTRY
              </th>
              <th>
                STATE
              </th>
              <th>
                CITY
              </th>
              <th>
                AREAS
              </th>
              <th />
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                <WithResponsiveMode
                  id="countryDrop"
                  onChanged={[Function]}
                  options={Array []}
                  placeHolder="Select a Country"
                />
              </td>
              <td>
                <WithResponsiveMode
                  id="stateDrop"
                  onChanged={[Function]}
                  options={Array []}
                  placeHolder="Select a State"
                />
              </td>
              <td>
                <WithResponsiveMode
                  id="cityDrop"
                  onChanged={[Function]}
                  options={Array []}
                  placeHolder="Select a City"
                />
              </td>
              <td>
                <WithResponsiveMode
                  multiSelect={true}
                  onChanged={[Function]}
                  options={Array []}
                  placeHolder="Select Areas"
                />
              </td>
              <td>
                <CustomizedDefaultButton
                  data-automation-id="resetButton"
                  onClick={[Function]}
                  text="Reset"
                />
              </td>
            </tr>
            <tr>
              <td
                colSpan="5"
                style={
                  Object {
                    "textAlign": "right",
                  }
                }
              >
                <CustomizedPrimaryButton
                  data-automation-id="addToTableButton"
                  id="addButton"
                  onClick={[Function]}
                  text="Add to Table"
                />
                <CustomizedPrimaryButton
                  data-automation-id="clearTableButton"
                  id="clearButton"
                  onClick={[Function]}
                  text="Clear Table"
                />
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        className="countryTable"
      >
        <table>
          <thead>
            <tr>
              <th>
                Country
              </th>
              <th>
                State
              </th>
              <th>
                City
              </th>
              <th>
                Areas
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <Territory />,
    "_forcedUpdate": false,
    "_instance": Territory {
      "addToTable": [Function],
      "clearTable": [Function],
      "componentDidMount": [Function],
      "context": Object {},
      "copyArray": [Function],
      "getAreasOfCity": [Function],
      "getCitiesOfState": [Function],
      "getStatesOfCountry": [Function],
      "onChangeCity": [Function],
      "onChangeCountry": [Function],
      "onChangeMultiSelect": [Function],
      "onChangeState": [Function],
      "props": Object {},
      "refs": Object {},
      "render": [Function],
      "resetForm": [Function],
      "state": Object {
        "cities": Array [],
        "count": 0,
        "countries": Array [],
        "rows": Array [],
        "selectedAreas": Array [],
        "selectedCity": "",
        "selectedCountry": "",
        "selectedItem": undefined,
        "selectedState": "",
        "states": Array [],
      },
      "updater": Updater {
        "_callbacks": Array [],
        "_renderer": [Circular],
      },
    },
    "_newState": null,
    "_rendered": <div
      className="appComponent"
    >
      <div
        className="countryForm"
      >
        <table>
          <thead>
            <tr>
              <th>
                COUNTRY
              </th>
              <th>
                STATE
              </th>
              <th>
                CITY
              </th>
              <th>
                AREAS
              </th>
              <th />
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                <WithResponsiveMode
                  id="countryDrop"
                  onChanged={[Function]}
                  options={Array []}
                  placeHolder="Select a Country"
                />
              </td>
              <td>
                <WithResponsiveMode
                  id="stateDrop"
                  onChanged={[Function]}
                  options={Array []}
                  placeHolder="Select a State"
                />
              </td>
              <td>
                <WithResponsiveMode
                  id="cityDrop"
                  onChanged={[Function]}
                  options={Array []}
                  placeHolder="Select a City"
                />
              </td>
              <td>
                <WithResponsiveMode
                  multiSelect={true}
                  onChanged={[Function]}
                  options={Array []}
                  placeHolder="Select Areas"
                />
              </td>
              <td>
                <CustomizedDefaultButton
                  data-automation-id="resetButton"
                  onClick={[Function]}
                  text="Reset"
                />
              </td>
            </tr>
            <tr>
              <td
                colSpan="5"
                style={
                  Object {
                    "textAlign": "right",
                  }
                }
              >
                <CustomizedPrimaryButton
                  data-automation-id="addToTableButton"
                  id="addButton"
                  onClick={[Function]}
                  text="Add to Table"
                />
                <CustomizedPrimaryButton
                  data-automation-id="clearTableButton"
                  id="clearButton"
                  onClick={[Function]}
                  text="Clear Table"
                />
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        className="countryTable"
      >
        <table>
          <thead>
            <tr>
              <th>
                Country
              </th>
              <th>
                State
              </th>
              <th>
                City
              </th>
              <th>
                Areas
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Territory />,
}
`;
